### PR TITLE
linux compatibility

### PR DESCRIPTION
**Pull request by themancalledjakob**
Hey Nick,

turns out that libxml2 is not on all distros, I somehow always had it installed so I didn't notice.
While I was at it, I also adjusted linux readme and put the code to switch from release/debug in a script.
Should make everything a smoother and cleaner.

Cheers,
Jakob

### CODE REVIEW

**Comment by NickHardeman** [comment]
Hi Jakob,
Thank you for the help with linux, I don't have a linux machine or much knowledge about it. 
I noticed you added a .so for libxml2, are there advantages to that? I noticed in ofxSvg that it uses a static lib [ofxSvg libxml2](https://github.com/openframeworks/openFrameworks/blob/b7b7d40ae68587201d1b10ea97f29afb956f8a70/addons/ofxSvg/addon_config.mk#L65)

Best,
Nick

**Comment by themancalledjakob** [comment]
Hey Nick,

good point there.

there are no relevant advantages to using .so.
It's just what I got by the package manager, and since it worked in all
linux distros I tested with, I put it in.

But since libxml2 is already there for ofxSvg, I'll quickly check if it
is compatible (should be), and if yes I can include it in exactly the
same way as they did and send another PR.
Then we'd also have other architectures.

Otherwise we might get issues when someone uses both ofxSvg and ofxFBX.

Best,
Jakob

**Comment by themancalledjakob** [comment]
Hey Nick,

TL;DR - updated with libxml2.a, tested for ofxSvg compatibility,
couldn't check if I broke osx/ios so i didn't send a PR

so, this was quite inspiring actually.

I had a closer look how they handle libraries in oF, and saw that I can
easily cannibalize the downloadLibs.sh script to handle libxml2 also
for ofxFBX. I included the modified version in the repo.

Anyways, now libxml2 is a static library, the same as in ofxSvg, and I
doublechecked that there are no issues when including both ofxFBX &
ofxSvg.

I didn't dare to send a pull request just yet, because this meant to
fiddle with addons_config.mk also for osx. Theoretically it should just
work, but I'd feel really bad breaking your build with a PR.
As a thought, it should be easy to download libxml2 with the script
also for osx/ios/windows/etc, but I have sourrounded myself with linux
machines only, and can't test if that works properly. Also I'm not sure
if you currently rely on libxml2 from brew or sth, or if it just runs
out of the box.

If you have time to test, and it doesn't break everything, let me know
:-)

https://github.com/themancalledjakob/ofxFBX

Cheers,
Jakob
